src/OneExplorer/OneExplorer.ts

```diff
@@ -176,7 +176,16 @@ abstract class Node {
 }
 
 class NodeFactory {
-  static create(type: NodeType, fpath: string, attr?: ArtifactAttr): Node {
+  static create(type: NodeType, fpath: string, attr?: ArtifactAttr): Node|undefined {
+    // WHY HIDDEN NODES ARE NOT TO BE CREATED?
+    //
+    // A 'TreeDataProvider<element>' expects every elements (Node) to be correspond to visible
+    // TreeItem, so let's not build hidden nodes.
+    if(attr?.canHide === true && OneTreeDataProvider.didHideExtra === true)
```

Comment:
Q) How does the code be executed if `attr` parameter is not given?

Reply:
It makes an error, thanks for pointing out :-D

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining